pkg/settings/settings.go

```diff
@@ -141,19 +144,30 @@ func (s Setting) Get() string {
 }
 
 func (s Setting) GetInt() int {
-	v := s.Get()
-	i, err := strconv.Atoi(v)
-	if err == nil {
-		return i
+	var (
+		i   int
+		err error
+		v   = s.Get()
+	)
+
+	if v != "" {
+		if i, err = strconv.Atoi(v); err == nil {
+			return i
+		}
+		logrus.Errorf("failed to parse setting %s=%s as int: %v", s.Name, v, err)
 	}
-	logrus.Errorf("failed to parse setting %s=%s as int: %v", s.Name, v, err)
-	i, err = strconv.Atoi(s.Default)
-	if err != nil {
+
+	if i, err = strconv.Atoi(s.Default); err != nil {
```

Comment:
return the default value 1800 instead of 0, if this unlucky failure happens

Reply:
if I failed to parse the default value, how do I return default value? 

If it could parse it correctly, it just return default value, but if it couldn't. The default value might be `1800ss`, I could't also return it as well.

Reply:
Or do you mean if I get the zero in controller, I just reassign 1800 to it?

Reply:
This function, in any abnormal case, should return the none-zero default value, at the moment it is 1800. The `0` means the file will be deleted right after it is generated.

This odd situation is caused by below line, the default value can only be a string, and you set the 1800 a direct string.

Maybe you can define 2 consts in the uitl, "1800" as a string and referred in below, and another 1800 as an int, which is used as the fallback value `GetInt() int`

`SupportBundleExpirationSeconds = NewSetting(SupportBundleExpirationSecondsSettingName, "1800")`

Those 2 consts are put together, and with a comment saying they need to changed at mean time.

Reply:
I think it's weird to put two const here. I think we could change the `setting.Def` type into `interface{}`. Then you could put anything you want instead of putting two const due to the constraint of `setting.Def`.

Reply:
```
if i, err = strconv.Atoi(s.Default); err != nil {
  return 30 // minutes
}
```

should be fine in this case

Reply:
If I understand correctly, it will affect other setting, right?

Reply:
I think we just keep it returning 0 instead of returning 30 here.

Reply:
But, if you return 30 in this function, it also break other features. For example, [support-bundle-timeout](https://docs.harvesterhci.io/v1.2/advanced/index#support-bundle-timeout) uses 0 as disabling. So, I suggest we need to enhance whole setting structure, otherwise it would be weird case here, like you said 0 is an invalid value. I think it's a different issue here.

Reply:
OK, then please replace 0 in the controller side, revert 0 to 30.

Reply:
Surely, it makes sense for me.